shell/platform/darwin/ios/framework/Source/FlutterEngine.mm

```diff
@@ -1402,6 +1402,10 @@ - (FlutterDartProject*)project {
   return _dartProject.get();
 }
 
+- (BOOL)isUsingImpeller {
+  return self.project.isImpellerEnabled;
```

Comment:
This is available via the shell, but there's nothing inherently wrong with exposing it on the project either.